src/metrology_apis/__init__.py

```diff
@@ -1,15 +1,66 @@
 """Metrology APIs."""
 
-from typing import Final, Protocol, Self, override, runtime_checkable
+from typing import (
+    Any,
+    Final,
+    Protocol,
+    Self,
+    override,
+    runtime_checkable,
+)
 
 import optype as op
 
 __version__: Final = "0.0.1.dev0"
 __all__ = ["__version__", "Dimension", "Quantity", "Unit"]
 
 
+type VT = Any
+type UT = Unit[Any]
+type QT = Quantity[Any, Any, Any]
+
+
+@runtime_checkable
+class MetrologyNamespace[Q: QT, V: VT, U: UT, D: Dimension](Protocol):
```

Comment:
```suggestion
class MetrologyNamespace[Q: QT = Any, V: VT = Any, U: UT = Any, D: Dimension = Any](Protocol):
```

or you could keep e.g. `Q` van `V` required:

```suggestion
class MetrologyNamespace[Q: QT, V: VT, U: UT = Any, D: Dimension = Any](Protocol):
```